connector/src/main/java/org/geysermc/connector/utils/SkinUtils.java

```diff
@@ -217,8 +217,17 @@ public static void handleBedrockSkin(PlayerEntity playerEntity, BedrockClientDat
             byte[] skinBytes = com.github.steveice10.mc.auth.util.Base64.decode(clientData.getSkinData().getBytes("UTF-8"));
             byte[] capeBytes = clientData.getCapeData();
 
-            byte[] geometryNameBytes = com.github.steveice10.mc.auth.util.Base64.decode(clientData.getGeometryName().getBytes("UTF-8"));
-            byte[] geometryBytes = com.github.steveice10.mc.auth.util.Base64.decode(clientData.getGeometryData().getBytes("UTF-8"));
+            byte[] geometryNameBytes;
+            byte[] geometryBytes;
+
+            if (clientData.getGeometryName() != null) {
+                geometryNameBytes = com.github.steveice10.mc.auth.util.Base64.decode(clientData.getGeometryName().getBytes("UTF-8"));
```

Comment:
Please use the built in java decoder:

`Base64.getDecoder().decode()`

Reply:
Actually, I’ve just noticed the code was already like that before your modifications...I’ve just woken up aha

Reply:
haha. I've tried not to change too much so left some things non-optimal.